File: services/gateway-service/src/main/java/com/codeconnect/gateway/application/dto/response/UserResponse.java
package com.codeconnect.gateway.application.dto.response;

import com.codeconnect.gateway.domain.enums.UserRole;
import com.codeconnect.gateway.domain.enums.UserStatus;

import java.time.Instant;

/**
 * Sanitized user representation returned to presentation callers.
 * Strictly never exposes password hashes.
 */
public record UserResponse(
    String id,
    String email,
    String displayName,
    UserRole role,
    UserStatus status,
    Instant createdAt
) {}

File: frontend/src/dto/response/UserResponse.ts
import { UserRole } from "@/domain/enums/UserRole";
import { UserStatus } from "@/domain/enums/UserStatus";

/**
 * Sanitized user response payload matching com.codeconnect.gateway.application.dto.response.UserResponse
 */
export interface UserResponse {
  readonly id: string;
  readonly email: string;
  readonly displayName: string;
  readonly role: UserRole;
  readonly status: UserStatus;
  readonly createdAt: string;
}
